tests/realtikvtest/importintotest/import_into_test.go

```diff
@@ -1239,3 +1239,26 @@ func (s *mockGCSSuite) TestBadCases() {
 	err := s.tk.QueryToErr(sql)
 	require.ErrorContains(s.T(), err, "panic occurred during import, please check log")
 }
+
+func (s *mockGCSSuite) TestImportIntoWithFK() {
+	content := []byte(`1,1
+	2,2`)
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{
+			BucketName: "foreign-key-test",
+			Name:       "child.csv",
+		},
+		Content: content,
+	})
+	tempDir := s.T().TempDir()
+	filePath := path.Join(tempDir, "file.csv")
+	s.NoError(os.WriteFile(filePath, content, 0644))
+	s.prepareAndUseDB("import_into")
+	s.tk.MustExec("create table parent (id int primary key);")
+	s.tk.MustExec("create table child (id int primary key, fk int, foreign key (fk) references parent(id));")
+	sql := fmt.Sprintf(`IMPORT INTO import_into.child
+		FROM 'gs://foreign-key-test/child.csv?endpoint=%s'`, gcsEndpoint)
+
+	// it should success even if the parent table is empty
+	s.tk.MustQuery(sql)
```

Comment:
maybe also check SELECT result

Reply:
Done.

Reply:
have you pushed code, i didn't see it

Reply:
> have you pushed code, i didn't see it

Oops. Network issue :facepalm:. Pushed now.